src/Karma/Hydrator.php

```diff
@@ -91,27 +102,38 @@ public function setSystemEnvironment($environment)
 
     public function hydrate($environment)
     {
-        $distFiles = $this->collectDistFiles();
+        $files = $this->collectFiles();
 
-        foreach($distFiles as $file)
+        foreach($files as $file)
         {
             $this->hydrateFile($file, $environment);
         }
 
         $this->info(sprintf(
            '%d files generated',
-            count($distFiles)
+            count($files)
         ));
     }
 
-    private function collectDistFiles()
+    private function collectFiles()
     {
-        return $this->finder->findFiles("~$this->suffix$~");
+        $pattern = sprintf('.*\%s$', $this->suffix);
+        if($this->nonDistFilesOverwriteAllowed === true)
+        {
+            $pattern = '.*';
+        }
+        
+        return $this->finder->findFiles(sprintf('~%s~', $pattern));
     }
 
     private function hydrateFile($file, $environment)
     {
-        $this->currentTargetFile = substr($file, 0, strlen($this->suffix) * -1);
+        $this->currentTargetFile = preg_replace(sprintf('~(.*)(\%s)$~', $this->suffix), '$1', $file);
+
+        if($this->nonDistFilesOverwriteAllowed)
+        {
+            $this->currentTargetFile = (new \SplFileInfo($this->currentTargetFile))->getFilename();
```

Comment:
This lines converts path/filename to filename
When targetPath config is enable, the generated files are copied in target dir. 
```
dir1
|_file1-dist
|_file2
dir2
|_file3-dist
```
Karma will generate the following files :
```
targetDir
|_file1-dist
|_file2
|_file3-dist
```
An exception is thrown in case of duplicated filenames in 2 separated folder
add 46228c3